app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,20 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+
+    respond_to do |format|
+      if @scenario.agents.update_all disabled: params[:scenario][:disabled]
```

Comment:
This is from better errors after I deliberately make it fail to access the console
```
>> params[:scenario][:disabled]
=> "true"
>> @scenario.agents
=> #<ActiveRecord::Associations::CollectionProxy [#<Agents::TwitterSearchAgent id: 38, user_id: 1, options: {"search"=>"freebandnames", "expected_update_period_in_days"=>"2"}, type: "Agents::TwitterSearchAgent", name: "Twitter Search Agent", schedule: "every_12h", events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:33", updated_at: "2016-05-16 18:54:39", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "c48ff0f1e8fe5359e8e84601fe618ae5", deactivated: false>, #<Agents::TwitterPublishAgent id: 37, user_id: 1, options: {"expected_update_period_in_days"=>"10", "message"=>"{{text}}"}, type: "Agents::TwitterPublishAgent", name: "Twitter Publish Agent", schedule: nil, events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:16", updated_at: "2016-05-16 18:59:40", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "b0cc289b0c160778033718bdd3f8dd6a", deactivated: false>, #<Agents::TwitterFavorites id: 36, user_id: 1, options: {"username"=>"tectonic", "number"=>"10", "history"=>"100", "expected_update_period_in_days"=>"2"}, type: "Agents::TwitterFavorites", name: "Twitter Favorites", schedule: "every_12h", events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:02", updated_at: "2016-05-16 19:03:26", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "03c480dfb76dd1a254309bee864ec7ae", deactivated: false>]>
>> @scenario.agents.update_all disabled: params[:scenario][:disabled]
=> 3
>> @scenario.agents
=> #<ActiveRecord::Associations::CollectionProxy [#<Agents::TwitterSearchAgent id: 38, user_id: 1, options: {"search"=>"freebandnames", "expected_update_period_in_days"=>"2"}, type: "Agents::TwitterSearchAgent", name: "Twitter Search Agent", schedule: "every_12h", events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:33", updated_at: "2016-05-16 18:54:39", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "c48ff0f1e8fe5359e8e84601fe618ae5", deactivated: false>, #<Agents::TwitterPublishAgent id: 37, user_id: 1, options: {"expected_update_period_in_days"=>"10", "message"=>"{{text}}"}, type: "Agents::TwitterPublishAgent", name: "Twitter Publish Agent", schedule: nil, events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:16", updated_at: "2016-05-16 18:59:40", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "b0cc289b0c160778033718bdd3f8dd6a", deactivated: false>, #<Agents::TwitterFavorites id: 36, user_id: 1, options: {"username"=>"tectonic", "number"=>"10", "history"=>"100", "expected_update_period_in_days"=>"2"}, type: "Agents::TwitterFavorites", name: "Twitter Favorites", schedule: "every_12h", events_count: 0, last_check_at: nil, last_receive_at: nil, last_checked_event_id: nil, created_at: "2016-05-11 21:25:02", updated_at: "2016-05-16 19:03:26", memory: {}, last_web_request_at: nil, keep_events_for: 0, last_event_at: nil, last_error_log_at: nil, propagate_immediately: false, disabled: false, service_id: 1, guid: "03c480dfb76dd1a254309bee864ec7ae", deactivated: false>]>
```

Notice that after the update all `disabled` is still false and during the `update_all` process nothing was every `committed`. 

I don't see any errors in my rails log

![screen shot 2016-05-18 at 11 13 33 am](https://cloud.githubusercontent.com/assets/1547934/15364210/a9518ea6-1ce9-11e6-9ef9-a1c98a45b55d.png)

I see a `before_save` call attach to the model which I commented out but it didn't go through. Also I recalled I can update a single record with `update_attributes` and had it successfully committed.